lib/vsc/utils/script_tools.py

```diff
@@ -210,3 +226,173 @@ def critical_exception_handler(self, tp, value, traceback):
+
+        # insert default timestamp value file based on name
+        if TIMESTAMP_FILE_OPTION in options:
+            tsopt = list(options[TIMESTAMP_FILE_OPTION])
+            tsopt[-1] = os.path.join(self.CACHE_DIR, "%s.timestamp" % self.name)
+            options[TIMESTAMP_FILE_OPTION] = tuple(tsopt)
+
+        options.update(self.CLI_OPTIONS)
+
+        if TIMESTAMP_FILE_OPTION not in options and self.TIMESTAMP_MANDATORY:
+            raise Exception("no mandatory %s option defined" % (TIMESTAMP_FILE_OPTION,))
+
+        return ExtendedSimpleOption(options)
+
+    def _update_kwargs_modified_since(self, kwargs):
+        """
+        Update dict kwargs in place with modified_since value based on start_timestamp and current_time
+        """
+        if 'modified_since' not in kwargs:
+            if self.start_timestamp is not None:
+                mtime = self.start_timestamp
+                if self.current_time is not None:
+                    mtime = '%s_%s' % (mtime, convert_to_unix_timestamp(self.current_time))
+                kwargs['modified_since'] = mtime
+
+    def warning(self, msg):
+        """
+        Convenience method that calls ExtendedSimpleOptions warning and exists with nagios warning exitcode
+        """
+        exit_from_errorcode(msg, 1)
+
+    def critical(self, msg):
+        """
+        Convenience method that calls ExtendedSimpleOptions critical and exists with nagios critical exitcode
+        """
+        exit_from_errorcode(msg, 2)
+
+    def critical_exception(self, msg, exception):
+        """
+        Convenience method: report exception and critical method
+        """
+        logging.exception("%s: %s", msg, exception)
+        exit_from_errorcode(msg, 2)
+
+    def do(self, dry_run):  #pylint: disable=unused-argument
+        """
+        Method to add actual work to do.
+        The method is executed in main method in a generic try/except/finally block
+        You can return something, that, when it evals to true, is considered fatal
+            self.start_timestamp has start time (i.e. either passed via commandline or
+                latest successful run from cache file)
+            self.current_time has current_time (to be used as next start_timestamp when all goes well)
+            self.options has options from commandline
+            self.thresholds can be used to pass the thresholds during epilogue
+        """
+        raise Exception("Not implemented")
+
+    def make_time(self):
+        """
+        Get start time (from commandline or cache), return current time
+        """
+        try:
+            (start_timestamp, current_time) = retrieve_timestamp_with_default(
+                getattr(self.options, TIMESTAMP_FILE_OPTION),
+                start_timestamp=self.options.start_timestamp,
+                default_timestamp=DEFAULT_TIMESTAMP,
+                delta=-MAX_RTT,  # make the default delta explicit, current_time = now - MAX_RTT seconds
```

Comment:
so the delat can probably be 0 if there is not external system to compare to. in the AP it has to be > 0, but not for everything. (obviously, won't matter much, it's only a few seconds anyway, maybe make this a bit clear somewhere)